locksmith/__tests__/controllers/purchaseController.test.ts

```diff
@@ -57,6 +51,18 @@ describe('Purchase Controller', () => {
     })
   })
 
+  beforeEach(() => {
+    jest.mock('../../src/payment/paymentProcessor', () => {
```

Comment:
https://jestjs.io/docs/en/jest-object#jestgenmockfrommodulemodulename plus https://jestjs.io/docs/en/mock-function-api.html#mockfnmockimplementationoncefn may be a better way to do this that fixes the skipped test